lms/services/document.py

```diff
@@ -0,0 +1,39 @@
+import re
+from dataclasses import dataclass
+from typing import Optional
+
+DOCUMENT_URL_REGEXES = {
+    # canvas://file/course/COURSE_ID/file_id/FILE_ID
+    "canvas": re.compile(
+        r"canvas:\/\/file\/course\/(?P<course_id>[^\/]*)\/file_id\/(?P<file_id>[^\/]*)"
+    ),
+    # blackboard://content-resource/<file_id>/ URLs
+    "blackboard": re.compile(r"blackboard:\/\/content-resource\/(?P<file_id>[^\/]*)\/"),
+    # d2l://file/course/COURSE_ID/file_id/FILE_ID.
+    "d2l": re.compile(
+        r"d2l:\/\/file\/course\/(?P<course_id>[^\/]*)\/file_id\/(?P<file_id>[^\/]*)\/"
+    ),
+}
+
+
+@dataclass
+class DocumentURLParts:
+    file_id: str
+    course_id: Optional[str] = None
+
+
+class DocumentService:
```

Comment:
This service is pretty basic right now.

The idea is to extend it to help provide a filename for the import/export.